### DISCUSSION TITLE
Support "create" request extensions

### DISCUSSION THREAD

**Discussion by mihalicyn** [libfuse/libfuse#867] [General]
Some time ago Miklos (@szmi) implemented a fuse protocol extension (for fuse requests which create new inodes) https://lore.kernel.org/all/20221110142020.191487-1-mszeredi@redhat.com/

Do we have any plan to support it from the libfuse side?

Miklos has also posted a libfuse patch in the LKML, but for some reason I can't see this change in the libfuse code.

I'm asking because right now I'm working on another one protocol extension [1], [2] that will be based on the same principle to support idmapped mounts in FUSE. So it would be great to have this change in libfuse. I'm ready to help with porting of it. But decided to create this issue in case if Miklos respond here.

[1] https://github.com/mihalicyn/linux/commits/fuse_idmapped_mounts
[2] https://github.com/mihalicyn/libfuse/commits/idmap_support
[3] https://github.com/mihalicyn/fuse-overlayfs/commits/idmap_support

**Comment by bsbernd** [comment]
I think Miklos did it for virtiofs only. And yes, we should add this to libfuse. I also need to  read your idmapped patches.
